### PR TITLE
Rsdk 2582 new vision service protos

### PR DESCRIPTION
**Pull request by purplenicole730**
[RSDK-2582](https://viam.atlassian.net/browse/RSDK-2582)

Changes in Vision Service introduced before scheduled release:
- generated protos locally that included many deletions of functions
- delete related functions in vision service client
- delete related tests and mocks

[RSDK-2582]: https://viam.atlassian.net/browse/RSDK-2582?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/test_vision_service.py [hunk 1]
```diff
@@ -16,12 +16,10 @@
 from viam.services.vision import (
     Detection,
     Classification,
-    VisModelConfig,
```
Comment (purplenicole730) [inline tests/test_vision_service.py:19]: All references to this dataclass are now gone. Is this dataclass still needed?

Comment (cheukt) [reply tests/test_vision_service.py:19]: can delete as well

File: src/viam/services/vision/client.py [hunk 2]
```diff
@@ -131,9 +73,7 @@ async def get_detections_from_camera(
         """
         if extra is None:
             extra = {}
-        request = GetDetectionsFromCameraRequest(
-            name=self.name, camera_name=camera_name, detector_name=detector_name, extra=dict_to_struct(extra)
```
Comment (cheukt) [inline src/viam/services/vision/client.py:135]: remove detector_name as an arg to this function above + update docs

File: src/viam/services/vision/client.py [hunk 3]
```diff
@@ -168,52 +108,11 @@ async def get_detections(
             width=image.width,
             height=image.height,
             mime_type=mime_type,
-            detector_name=detector_name,
```
Comment (cheukt) [inline src/viam/services/vision/client.py:171]: remove detector_name as an arg to this function above + update docs

File: src/viam/services/vision/client.py [hunk 4]
```diff
@@ -235,9 +134,7 @@ async def get_classifications_from_camera(
         """
         if extra is None:
             extra = {}
-        request = GetClassificationsFromCameraRequest(
-            name=self.name, camera_name=camera_name, classifier_name=classifier_name, n=count, extra=dict_to_struct(extra)
-        )
+        request = GetClassificationsFromCameraRequest(name=self.name, camera_name=camera_name, n=count, extra=dict_to_struct(extra))
```
Comment (cheukt) [inline src/viam/services/vision/client.py:137]: remove classifier_name as an arg to this function above + update docs

File: src/viam/services/vision/client.py [hunk 5]
```diff
@@ -368,7 +205,6 @@ async def get_object_point_clouds(
         request = GetObjectPointCloudsRequest(
             name=self.name,
             camera_name=camera_name,
-            segmenter_name=segmenter_name,
```
Comment (cheukt) [inline src/viam/services/vision/client.py:371]: remove segmenter_name as an arg to this function above + update docs

**Review by cheukt — CHANGES_REQUESTED** [review]
if a parameter is no longer needed in the proto, we can get rid of it from the wrapper as well

**Review by cheukt — APPROVED** [review]
LGTM but remember to remove `VisModelType` and `VisModelConfig` as well

**Review by stuqdog — APPROVED** [review]
lgtm!

File: src/viam/services/vision/client.py [hunk 6]
```diff
@@ -253,7 +125,6 @@ async def get_classifications(
 
         Args:
             image (Image): The image to get detections from
-            classifier_name (str): The name of the detector to use for detection
```
Comment (bhaney) [inline src/viam/services/vision/client.py:128]: It looks like get_classifications is missing the "count" (also known as "n") argument from the api:
https://github.com/viamrobotics/api/blob/RSDK-2578-new-vision-service/proto/viam/service/vision/v1/vision.proto#L115

get_classifications_from_camera has it, though

Comment (purplenicole730) [reply src/viam/services/vision/client.py:129]: fixed!

**Pull request merged**